lib/ansible/modules/infrastructure/stacki/stacki_host.py

```diff
@@ -0,0 +1,332 @@
+    
+        self.stack_sync()
+    
+        result['changed'] = changed
+        result['stdout'] = "api call successful".rstrip("\r\n")
+        result['rc'] = rc
+   
+    def stack_add_interface(self):
+
+        data['cmd'] = "add host interface {} interface={} ip={} network={} mac={} default=true"\
+            .format(self.hostname, self.prim_intf, self.prim_intf_ip, self.network, self.prim_intf_mac)
+        stack_r = client.post(self.endpoint, data=json.dumps(data), headers=self.header)
+        rc = stack_r.status_code
+    
+        if rc == 200:
+            rc = 0
+        else:
+            self.module.fail_json(changed=False, result=stack_r.text)
+    
+
+    def stack_add(self, result):
+    
+        data            = dict()
+        rc              = None
+        changed         = False
+    
+        data['cmd'] = "add host {} rack={} rank={} appliance={}"\
+            .format(self.hostname, self.rack, self.rank, self.appliance)
+        stack_r = client.post(self.endpoint, data=json.dumps(data), headers=self.header)
+        rc = stack_r.status_code
+    
+        if rc == 200:
+            rc = 0
+        else:
+            self.module.fail_json(changed=False, result=stack_r.text)
+    
+        self.stack_sync()
+    
+        result['changed'] = changed
+        result['stdout'] = "api call successful".rstrip("\r\n")
+        result['rc'] = rc
+    
+    
+    def stack_remove(self, result):
+    
+        data            = dict()
+        rc              = None
+    
+        data['cmd'] = "remove host {}"\
+            .format(self.hostname)
+        stack_r = client.post(self.endpoint, data=json.dumps(data), headers=self.header)
+        rc = stack_r.status_code
+    
+        if rc == 200:
+            rc = 0
+        else:
+            self.module.fail_json(changed=False, result=stack_r.text)
+    
+        self.stack_sync()
+    
+        result['changed'] = True
+        result['stdout'] = "api call successful".rstrip("\r\n")
+        result['rc'] = rc
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            name=dict(required=True, type='str'),
+            rack=dict(required=False, type='int', default=0),
+            rank=dict(required=False, type='int', default=0),
+            appliance=dict(required=False, type='str', default='backend'),
+            prim_intf=dict(required=False, type='str', default=None),
+            prim_intf_ip=dict(required=False, type='str', default=None),
+            network=dict(required=False, type='str', default='private'),
+            prim_intf_mac=dict(required=False, type='str', default=None),
+            stacki_user=dict(required=True, type='str', default=os.environ.get('stacki_user')),
+            stacki_password=dict(required=True, no_log=True, type='str', default=os.environ.get('stacki_password')),
```

Comment:
You should mention these these env variables in the `description:`, e.g.

"If not specified the environment variable C(stacki_password) is used instead."